lib/streamlit/runtime/legacy_caching/caching.py

```diff
@@ -66,6 +68,74 @@
 _CacheEntry = namedtuple("_CacheEntry", ["value", "hash"])
 _DiskCacheEntry = namedtuple("_DiskCacheEntry", ["value"])
 
+# When we show the "st.cache is deprecated" warning, we make a recommendation about which new
+# cache decorator to switch to for the following data types:
+NEW_CACHE_FUNC_RECOMMENDATIONS: Dict[str, CacheType] = {
+    # cache_data recommendations:
+    "str": CacheType.DATA,
+    "float": CacheType.DATA,
+    "int": CacheType.DATA,
+    "bytes": CacheType.DATA,
+    "bool": CacheType.DATA,
+    "datetime.datetime": CacheType.DATA,
+    "pandas.DataFrame": CacheType.DATA,
+    "pandas.Series": CacheType.DATA,
+    "numpy.ndarray": CacheType.DATA,
+    "numpy.float64": CacheType.DATA,
+    "numpy.int64": CacheType.DATA,
```

Comment:
Minor thing, but I just wondering what is the logic for choosing these particular types for recommendation: why e.g. we not include `numpy.int16`, `numpy.float32` etc?  

I am not an expert, mentioning @LukasMasuch here. Lukas, maybe you know an elegant way to enumerate all appropriate NumPy types here.

Reply:
@jrieke chose the types, but I agree: let's include all numpy types in there! I'll update